crates/rpc_client/src/lib.rs

```diff
@@ -0,0 +1,314 @@
+use rethnet_eth::{transaction::Transaction, H256};
+
+// provide interfaces for all of the client functionality depended on by the existing Hardhat
+// Network logic, specifically
+// packages/hardhat-core/src/internal/hardhat-network/provider/fork/{ForkBlockchain,
+// ForkStateManager}.ts
+// and even more specifically, all of the methods shown in these excerpts:
+//
+//      const remote = await this._jsonRpcClient.getTransactionByHash(
+//    const remote = await this._jsonRpcClient.getTransactionReceipt(
+//      const remoteLogs = await this._jsonRpcClient.getLogs({
+//    const rpcBlock = await this._jsonRpcClient.getBlockByHash(blockHash, true);
+//    const rpcBlock = await this._jsonRpcClient.getBlockByNumber(
+//      const noncePromise = this._jsonRpcClient.getTransactionCount(
+//      const accountData = await this._jsonRpcClient.getAccountData(
+//    const remoteValue = await this._jsonRpcClient.getStorageAt(
+//
+// TODO: do more than just get-tx-by-hash.
+
+mod jsonrpc_types {
```

Comment:
You are more than welcome to keep using the dependency if this is a 1-to-1 copy

Reply:
The dependency is old and unmaintained ("archived" even, iirc), so I'm not comfortable depending on it if we don't have to. I actually already cut most of the dependency out by hand-rolling the JSON request string ([here](https://github.com/NomicFoundation/hardhat/pull/3383/files#diff-5936ebb05158585965a0600996c62b5b34055eb5cd63f356d8ac1ae6edab885eR152)). And the request string is in a sense validated by the remote node, so we don't need the type checking as much for the request anyways. However, the response validation is essential, and it was easy enough to inline this relatively small amount of code to get it, and completely cut the external dependency, so I went with this.

Reply:
Sounds good